### ISSUE TITLE
Specifying timestep in FAH YAML leads to error

### ISSUE THREAD

**Issue by jchodera** [choderalab/perses#728]
Manually specifying a timestep in the FAH YAML file:
```YAML
protein_pdb: protein.pdb
ligand_file: DOCKING_4MDS_POSES.sdf
old_ligand_index: 0
new_ligand_index: 1
forcefield_files:
    - amber/ff14SB.xml # ff14SB protein force field
    - amber/tip3p_standard.xml # TIP3P and recommended monovalent ion parameters
    - amber/tip3p_HFE_multivalent.xml # for divalent ions
    - amber/phosaa10.xml # HANDLES THE TPO
# don't constrain bonds to hydrogen to work around pathological slowdown with CCMA
remove_constraints: not water 
atom_expression:
    - IntType
bond_expression:
    - DefaultBonds
small_molecule_forcefield:
    - openff-1.2.0
phase_project_ids:
    complex: 13424
    solvent: 13425
    vacuum: vacuum-sprint3
ionic_strength: 0.070 # molar, to match experimental conditions
timestep: 2.0 # 2 fs
```
leads to this error:
```
2020-08-26 23:29:31,741:(0.00s):root:Setting up phase complex
Traceback (most recent call last):
  File "/home/chodera/miniconda/envs/perses/lib/python3.7/site-packages/simtk/unit/quantity.py", line 202, in __getattr__
    ret_val = getattr(self._value, attribute)
AttributeError: 'float' object has no attribute 'exp'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/chodera/miniconda/envs/perses/bin/perses-fah", line 33, in <module>
    sys.exit(load_entry_point('perses==0.8.1.dev0', 'console_scripts', 'perses-fah')())
  File "/home/chodera/miniconda/envs/perses/lib/python3.7/site-packages/perses-0.8.1.dev0-py3.7.egg/perses/app/fah_generator.py", line 469, in run
    run_neq_fah_setup(**setup_options)
  File "/home/chodera/miniconda/envs/perses/lib/python3.7/site-packages/perses-0.8.1.dev0-py3.7.egg/perses/app/fah_generator.py", line 374, in run_neq_fah_setup
    integrator = make_neq_integrator(**setup_options)
  File "/home/chodera/miniconda/envs/perses/lib/python3.7/site-packages/perses-0.8.1.dev0-py3.7.egg/perses/app/fah_generator.py", line 81, in make_neq_integrator
    integrator = PeriodicNonequilibriumIntegrator(alchemical_functions, nsteps_eq, nsteps_neq, neq_splitting, timestep=timestep)
  File "/home/chodera/miniconda/envs/perses/lib/python3.7/site-packages/openmmtools/integrators.py", line 1980, in __init__
    super().__init__(alchemical_functions, splitting, nsteps_neq=nsteps_neq, **kwargs)
  File "/home/chodera/miniconda/envs/perses/lib/python3.7/site-packages/openmmtools/integrators.py", line 1783, in __init__
    super(AlchemicalNonequilibriumLangevinIntegrator, self).__init__(*args, **kwargs)
  File "/home/chodera/miniconda/envs/perses/lib/python3.7/site-packages/openmmtools/integrators.py", line 1578, in __init__
    super(NonequilibriumLangevinIntegrator, self).__init__(*args, **kwargs)
  File "/home/chodera/miniconda/envs/perses/lib/python3.7/site-packages/openmmtools/integrators.py", line 1138, in __init__
    self.addGlobalVariable("a", np.exp(-gamma * h))
TypeError: loop of ufunc does not support argument 0 of type Quantity which has no callable exp method
```

**Comment by jchodera** [comment]
I believe the issue here is that `run_setup()` assumes the `timestep` option will be a `float` but `make_neq_integrator` assumes the `timestep` argument will have units. 

@mcwitt might be able to give pointers on introducing explicit typing to avoid this particular issue.

Currently, there is no consistent way to specify a valid timestep through the YAML file, so I will just change all the defaults locally.

**Comment by jchodera** [comment]
@Hannah @dom.rufa: Even changing the timestep defaults in the optional argument function definitions and omitting `timestep` from the YAML file results in incorrect units:
```
2020-08-27 00:48:30,536:(0.00s):root:   timestep: 2.0 fs**2.
```
It is not safe to start these jobs on FAH if they've been equilibrated with an incorrect timestep, since it is likely many of them will fail. This likely explains our high NaN rate as well.

Once you have fixed this, I'll try again. For now, I'll wait until we can get this to work.

That puts the earliest start for Sprint 3 at Friday late at night. We should cancel the Friday morning meeting.

**Comment by jchodera** [comment]
(accidentally hit the wrong button)

**Comment by hannahbrucemacdonald** [comment]
Thanks -- I will fix this now. A decent percentage of our issues seem to come from the definition of variables/constants, and is something that we tried to fix a while ago but has been constantly wrapped up in the API changes.

I would propose that after fixing this, I just write something more robust for handling defaults/simulation variables/yaml inputs.

**Comment by hannahbrucemacdonald** [comment]
Closed with #729
